detection_rules/packaging.py

```diff
@@ -229,6 +229,9 @@ def from_config(cls, config: dict = None, verbose: bool = False) -> 'Package':
 
         rules = all_rules.filter(lambda r: filter_rule(r, rule_filter, exclude_fields))
 
+        # add back in deprecated fields
+        rules.deprecated = all_rules.deprecated
```

Comment:
unrelated bug: the `RuleCollection.filter` method strips deprecated rules, so this manually adds them back